lib/gateways/planGateway.js

```diff
@@ -40,6 +53,33 @@ export default class PlanGateway {
 
     const result = await this.client.query(request).promise();
     if (result.Items.length === 0) return null;
-    return new Plan(result.Items[0]);
+    return fromDynamoModel(result.Items[0]);
+  }
+
+  async find({ firstName, lastName, systemIds = [] }) {
+    if (!firstName) throw new Error('first name cannot be null.');
+    if (!lastName) throw new Error('last name cannot be null.');
+
+    const request = {
+      TableName: this.tableName,
+      IndexName: 'name_idx',
+      KeyConditionExpression: 'lastName = :l and firstName = :f',
+      ExpressionAttributeValues: {
+        ':l': lastName.toLowerCase(),
+        ':f': firstName.toLowerCase()
+      }
+    };
+
+    const { Items } = await this.client.query(request).promise();
+
+    return Promise.all(
```

Comment:
I believe that both filter and map are synchronous so will not require a Promise.all?

Reply:
Haha, yep because this index is just a projection of `systemIds` and name.